src/aggregator/client/writer.go

```diff
@@ -267,10 +267,10 @@ func (w *writer) encodeUntimedWithLock(
 	}
 
 	if encodeErr != nil {
-		w.log.WithFields(
-			log.NewField("metric", metricUnion),
-			log.NewField("metadatas", metadatas),
-			log.NewErrField(encodeErr),
+		w.log.With(
+			zap.Any("metric", metricUnion),
```

Comment:
I didn't look too much into the zap docs, but is `.Any()` the best for interfaces like this?

Reply:
Yeah. `Any()` basically will try to use the cheapest encoder it can (so int if the value is an int, etc), and then fall back to the most expensive Reflection based encoder for structs. Will be same cost or cheaper than the implicit `fmt.Sprintf(`%v`, struct{})` call we used to have.